### PR TITLE
feat(modal): Adds `defaultContainer` to `$modalProvider`

### PR DESCRIPTION
**Pull request by Lalem001**
Includes basic tests, with adjustments to toHaveModalsOpen and toHaveBackdrop.

Fixes: #2688 
Replaces PR: #2942

### CODE REVIEW

**Comment by marlonmleite** [comment]
+1

File: src/modal/modal.js [hunk 1]
```diff
@@ -454,6 +456,7 @@ angular.module('ui.bootstrap.modal', [])
                 backdropClass: modalOptions.backdropClass,
                 windowClass: modalOptions.windowClass,
                 windowTemplateUrl: modalOptions.windowTemplateUrl,
+                container: $modalProvider.defaultContainer,
```
Comment (wesleycho) [inline src/modal/modal.js]: Looking at the original issue this PR references, any particular reason for the choice of `container` over `appendTo`? I think the latter makes more sense semantically, but am open to other options.

File: src/modal/test/modal.spec.js [hunk 2]
```diff
@@ -73,9 +73,11 @@ describe('$modal', function () {
       toHaveModalOpenWithContent: function(util, customEqualityTesters) {
         return {
           compare: function(actual, content, selector) {
-            var contentToCompare, modalDomEls = actual.find('body > div.modal > div.modal-dialog > div.modal-content');
+            var container = (actual === $document) ? actual.find('body') : angular.element(actual);
```
Comment (wesleycho) [inline src/modal/test/modal.spec.js]: Shouldn't `actual` already be a jqLite wrapped element? I believe `angular.element(actual)` should just be `actual` here.

File: src/modal/test/modal.spec.js [hunk 3]
```diff
@@ -94,7 +96,9 @@ describe('$modal', function () {
       toHaveModalsOpen: function(util, customEqualityTesters) {
         return {
           compare: function(actual, expected) {
-            var modalDomEls = actual.find('body > div.modal');
+            var container = (actual === $document) ? actual.find('body') : angular.element(actual);
```
Comment (wesleycho) [inline src/modal/test/modal.spec.js]: Ditto

File: src/modal/modal.js (hunk 1 above)
Comment (Lalem001) [inline src/modal/modal.js]: `container` is just coming off of my use of `defaultContainer`. Would you want `appendTo` and `defaultContainer`? Or change both to `appendTo` and `defaultAppendTo`?

Or perhaps something else entirely?

Comment (wesleycho) [inline src/modal/modal.js]: It should be `appendTo` and there should be no `defaultContainer` reference.

**Comment by wesleycho** [comment]
Should be noted, we now have an `appendTo` feature via 16d854ca4c70aefb6b03faeb2a20105ce2bea8ec - this feature should interop with the new implementation. Given that it takes a DOM node in the $modal service, I think this feature then becomes invalid. If one needs a sensible default, I recommend creating a modal abstraction on top of $modal to pass the desired defaults.

**Pull request closed without merging**